modules/util/dimensions.js

```diff
@@ -17,15 +17,27 @@ export function utilGetDimensions(selection, force) {
 
 export function utilSetDimensions(selection, dimensions) {
     if (!selection || selection.empty()) {
-        return selection;
+        selection = {
```

Comment:
Interesting choice to mock a d3-selection here!
An easier way is to do something like

`selection = selection || d3_select(null)`  which will fallback to making a _real_ d3 selection with nothing in it.  Then you can use that selection elsewhere without the code blowing up.